api/pkg/transformer/symbol/time.go

```diff
@@ -0,0 +1,146 @@
+}
+
+// IsWeekend check wheter given timestamps falls in weekend, given timestamp and timezone
+// timestamp can be:
+// - Json path string
+// - Slice / gota.Series
+// - int64 value
+func (sr Registry) IsWeekend(timestamp interface{}, timezone string) interface{} {
+	timeFn := func(ts int64, tz *time.Location) interface{} {
+		return function.IsWeekend(ts, tz)
+	}
+	return sr.processTimestampFunction(timestamp, timezone, timeFn)
+}
+
+// FormatTimestamp will convert timestamp into formatted date string
+// timestamp can be:
+// - Json path string
+// - Slice / gota.Series
+// - int64 value
+func (sr Registry) FormatTimestamp(timestamp interface{}, timezone, format string) interface{} {
+	timeFn := func(ts int64, tz *time.Location) interface{} {
+		return function.FormatTimestamp(ts, tz, format)
+	}
+	return sr.processTimestampFunction(timestamp, timezone, timeFn)
+}
+
+// DayOfWeek will return number represent day in a week, given timestamp and timezone
+// SUNDAY(0), MONDAY(1), TUESDAY(2), WEDNESDAY(3), THURSDAY(4), FRIDAY(5), SATURDAY(6)
+// timestamp can be:
+// - Json path string
+// - Slice / gota.Series
+// - int64 value
+func (sr Registry) DayOfWeek(timestamp interface{}, timezone string) interface{} {
+	timeFn := func(ts int64, tz *time.Location) interface{} {
+		return function.DayOfWeek(ts, tz)
+	}
+	return sr.processTimestampFunction(timestamp, timezone, timeFn)
+}
+
+func (sr Registry) processTimestampFunction(timestamps interface{}, timezone string, timeTransformerFn func(timestamp int64, tz *time.Location) interface{}) interface{} {
+	timeLocation, err := time.LoadLocation(timezone)
+	if err != nil {
+		panic(err)
+	}
+
+	ts, err := sr.evalArg(timestamps)
+	if err != nil {
+		panic(err)
+	}
+	timestampVals := reflect.ValueOf(ts)
+	switch timestampVals.Kind() {
+	case reflect.Slice:
+		var values []interface{}
+		for idx := 0; idx < timestampVals.Len(); idx++ {
+			val := timestampVals.Index(idx)
+			tsInt64, err := converter.ToInt64(val.Interface())
+			if err != nil {
+				panic(err)
+			}
+			values = append(values, timeTransformerFn(tsInt64, timeLocation))
+		}
+		return values
+	default:
+		tsInt64, err := converter.ToInt64(ts)
+		if err != nil {
+			panic(err)
+		}
+		return timeTransformerFn(tsInt64, timeLocation)
+	}
+}
+
+// ParseDateTime converts datetime given with specified format layout (e.g. RFC3339) into time.
+// It uses Golang provided time format layout (https://pkg.go.dev/time#pkg-constants).
+//
+// Examples:
+// Request JSON: {"booking_time": "2021-12-01 11:30:00"}
+// Expression: ParseDateTime($.booking_time, "2006-01-02 15:04:05")
+// Output: 2021-12-01 11:30:00 UTC
+//
+// Request JSON: {"booking_time": "2021-12-01 11:30:00", "timezone": "Asia/Jakarta"}
+// Expression: ParseDateTime($.booking_time, "2006-01-02 15:04:05", $.timezone)
+// Output: 2021-12-01 11:30:00 +07:00
+func (sr Registry) ParseDateTime(datetime interface{}, format string, timezone ...string) interface{} {
```

Comment:
Why is the `timezone` variadic?

Reply:
It seems that ParseDateTime can follow same implementation using `processTimestampFunction`

Reply:
> Why is the timezone variadic?

As shown in the example in the comment, it allows users to ParseDateTime without considering any timezone. I prefer the user to ignore the timezone field instead of passing an empty string ("").

> It seems that ParseDateTime can follow same implementation using processTimestampFunction

No, it cannot. `processTimestampFunction` assumes timestamp which is an integer. Date time is a string.

Reply:
> As shown in the example in the comment, it allows users to ParseDateTime without considering any timezone. I prefer the user to ignore the timezone field instead of passing an empty string ("").


Can we make it consistent with other expression then?

Reply:
Which expression ya?
If we want to align with others, it's better to not have location here as variadic

Reply:
Other time-related expression that @tiopramayudi created. Yes, either the others follow this or the other way around. But please be consistent.